src/values.cpp

```diff
@@ -641,6 +645,27 @@ auto operator<<(std::ostream& os, const UnaryOrigin& self) -> std::ostream& {
     return os;
 }
 
+// struct NaryOrigin
+auto operator==(const NaryOrigin& lhs, const NaryOrigin& rhs) noexcept -> bool {
+    return lhs.values == rhs.values && lhs.location == rhs.location;
+}
+auto operator!=(const NaryOrigin& lhs, const NaryOrigin& rhs) noexcept -> bool {
+    return !(lhs == rhs);
+}
+auto operator<<(std::ostream& os, const NaryOrigin& self) -> std::ostream& {
+    os << "UnaryOrigin{ "
+       << ".values = " << self.values << ", "
+       << ".location = ";
+    if (self.location) {
+        os << self.location.value();
+    } else {
+        os << "nullopt";
+    }
+    os << ", .reverse = " << reinterpret_cast<void*>(self.reverse.target<UnaryOrigin::ReverseFn>());
```

Comment:
```suggestion
    os << ", .reverse = " << reinterpret_cast<void*>(self.reverse.target<NaryOrigin::ReverseFn>());
```

If the types don't match `std::functioon.terget<T>()` will always return a nullptr.